dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -65,12 +66,17 @@
     private final ApacheHttpClientChannels.CloseableClient client;
     private final BaseUrl baseUrl;
     private final ResponseLeakDetector responseLeakDetector;
+    private final OptionalInt uriIndexForInstrumentation;
 
     ApacheHttpClientBlockingChannel(
-            ApacheHttpClientChannels.CloseableClient client, URL baseUrl, ResponseLeakDetector responseLeakDetector) {
+            ApacheHttpClientChannels.CloseableClient client,
+            URL baseUrl,
+            ResponseLeakDetector responseLeakDetector,
+            OptionalInt uriIndexForInstrumentation) {
```

Comment:
Would Map<String,String> be a bit more generic and mean we could pass more data to this layer later should we wish to?

Reply:
What if this was encapsulated in an attachment on the request that is set/unset by another channel before this channel and then this channel pulls it out from the attachment and attaches as args to the exception here? Maybe then you wouldn't even need to change the creation API here.

Reply:
The args object provided by the new channel factory gives us just this, in a strongly typed way. We can always add additional optional fields to it.

I'd prefer not to use request attachments because a request instance may be reused, and I'm sure I'd eventually break the code which unsets data, and confuse myself into oblivion :-)